modules/core/src/effects/lighting-effect.js

```diff
@@ -1,7 +1,9 @@
 import {AmbientLight, PointLight, DirectionalLight} from '@luma.gl/core';
+import CameraLight from './camera-light';
 import Effect from '../lib/effect';
 import {projectPosition} from '../shaderlib/project/project-functions';
 import {COORDINATE_SYSTEM} from '../lib';
+import {getUniformsFromViewport} from '../shaderlib/project/viewport-uniforms';
```

Comment:
My main thought here is that importing `getUniformsFromViewport` here "exposes" a lot of complexity from the project module that we don't need here.

Could we export an additional, more focused function like `projectPosition` from the project module instead?

Ultimately, there should be a limited set of pre-projection operations that lighting modules need to do in JS.

Reply:
Here is not focusing on the 'projectPosition', but on 'getCameraPositionFromViewport'. Should I expose an API like that? Right now the camera position and coordination system switching logic is pretty complicated in our viewport-uniform module, initially I don't want to duplicate that kind of code in another function, that's why I just use the regular 'getUniformsFromViewport' API.

Reply:
Per discussion offline, will create a new API 'getCameraPosition' to serve camera light purpose.